src/khoj/routers/api_chat.py

```diff
@@ -150,6 +209,66 @@ async def clear_chat_history(
     return {"status": "ok", "message": "Conversation history cleared"}
 
 
+@api_chat.post("/public/fork")
+@requires(["authenticated"])
+def fork_public_conversation(
+    request: Request,
+    common: CommonQueryParams,
+    public_conversation_slug: str,
+):
+    user = request.user.object
+
+    # Load Conversation History
+    public_conversation = PublicConversationAdapters.get_public_conversation_by_slug(public_conversation_slug)
+
+    # Duplicate Conversation History to Public Conversation
+    ConversationAdapters.create_conversation_from_public_conversation(
+        user, public_conversation, request.user.client_app
+    )
+
+    chat_metadata = {"forked_conversation": public_conversation.slug}
+
+    update_telemetry_state(
+        request=request,
+        telemetry_type="api",
+        api="fork_public_conversation",
+        **common.__dict__,
+        metadata=chat_metadata,
+    )
+
+    redirect_uri = str(request.app.url_path_for("chat_page"))
+
+    return Response(status_code=200, content=json.dumps({"status": "ok", "next_url": redirect_uri}))
+
+
+@api_chat.post("/public/duplicate")
```

Comment:
The /duplicate name of the api endpoints seems unclear as to what it does IMO. The directionality of the duplication isn't specified, so it could mean duplicate from private to public or public to private. Maybe just POST "/public" should be good enough?

```suggestion
@api_chat.post("/public")
```